src/libraries/System.Text.Json/gen/Reflection/TypeWrapper.cs

```diff
@@ -311,9 +311,12 @@ public override FieldInfo[] GetFields(BindingFlags bindingAttr)
                         // we want a static field and this is not static
                         (BindingFlags.Static & bindingAttr) != 0 && !fieldSymbol.IsStatic ||
                         // we want an instance field and this is static or a constant
-                        (BindingFlags.Instance & bindingAttr) != 0 && (fieldSymbol.IsStatic || fieldSymbol.IsConst) ||
+                        (BindingFlags.Instance & bindingAttr) != 0 && (fieldSymbol.IsStatic || fieldSymbol.IsConst)
+#if ROSLYN4_0_OR_GREATER
                         // symbol represents an explicitly named tuple element
-                        fieldSymbol.IsExplicitlyNamedTupleElement)
+                        || fieldSymbol.IsExplicitlyNamedTupleElement
```

Comment:
I wish could avoid this. Effectively it will break VS19 sourcegen for any type containing value tuples. Would it be possible to consider potential workarounds for 3.9 or upgrade to 3.11?

One potential workaround would be to check if `fieldSymbol.Name` does not match the `Item[1-7]`.

Reply:
This might also provide hints on how the property could be derived:

https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Tuples/TupleErrorFieldSymbol.cs#L94

Reply:
@jcouv @chsienki - any thoughts here? Is there a way we can simulate the `fieldSymbol.IsExplicitlyNamedTupleElement` logic while targeting Roslyn 3.9? Or should we just target 3.10 which has this API?

Reply:
In an offline discussion with @jaredpar, I decided to upgrade to 3.11. If anyone has concerns/feedback, we can revisit this decision.